src/Components/Testimonials/TestimonialsForm.js

```diff
@@ -1,48 +1,200 @@
+import React, { useState, useEffect } from 'react';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import { useDropzone } from 'react-dropzone';
+import { useFormik } from 'formik';
+import { TextField, Box, Button, Alert, Typography } from '@mui/material';
+import axios from 'axios';
+
+import '../../Styles/TestimonialsFormStyles.css';
 import '../FormStyles.css';
 
-const TestimonialForm = () => {
-  const [initialValues, setInitialValues] = useState({
-    name: '',
-    description: '',
-  });
+const TestimonialForm = ({ id }) => {
+  const [description, setDescription] = useState('');
+  const [image, setImage] = useState('');
+  const [imageError, setImageError] = useState(false);
+
+  const getTestimonials = () => {
+    return {
+      name: 'Testimonial testings',
+      description: 'This is a mockup testimonial',
+      image: '',
+    };
+  };
+
+  useEffect(() => {
+    if (id) {
+      const resp = getTestimonials();
 
-  const handleChange = (e) => {
-    if (e.target.name === 'name') {
-      setInitialValues({ ...initialValues, name: e.target.value });
+      formik.values.name = resp.name;
+      setDescription(resp.description);
     }
-    if (e.target.name === 'description') {
-      setInitialValues({ ...initialValues, description: e.target.value });
+  }, []);
+
+  const validate = (values) => {
+    const errors = {};
+
+    if (!values.name) {
+      errors.name = 'El nombre es requerido';
+    } else if (values.name.length < 4) {
+      errors.name = 'El nombre debe contener al menos 4 caracteres';
+    }
+    if (!description) {
+      errors.description = 'La descripción es requerida';
     }
+    if (!image) {
+      errors.image = 'La imagen es requerida';
+    }
+
+    return errors;
   };
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    console.log(initialValues);
+  const formik = useFormik({
+    initialValues: {
+      name: '',
+      description: '',
+      image: '',
+    },
+    validate,
+    onSubmit: (values) => handleSubmit(values),
+  });
+
+  const handleCKeditorChange = (e, editor) => {
+    const data = editor.getData();
+
+    setDescription(data);
   };
 
+  const [files, setFiles] = useState([]);
+  const { getRootProps, getInputProps, fileRejections } = useDropzone({
+    multiple: false,
+    maxFiles: 1,
+    accept: 'image/jpeg, image/png',
+    onDrop: (acceptedFiles, fileRejections) => {
+      setFiles(
+        acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
+      );
+
+      if (fileRejections.length === 0) {
+        const reader = new FileReader();
+
+        reader.readAsDataURL(acceptedFiles[0]);
+        reader.onload = () => {
+          const base64 = reader.result;
+
+          setImage(base64);
+        };
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (fileRejections.length > 0) {
```

Comment:
please move this validation into a function in utils:
```
const listHasValues = (list) => list.length > 0;
```
then use it here:
```
if(listHasValues(fileRejections) {
    setImageError(true);
    return;
}
setImageError(false)
```

Reply:
Also move this validation out of this useEffect, create a function and then call the function here.
ex:
```
const validateImage = () => {
if(listHasValues(fileRejections) {
    setImageError(true);
    return;
}
setImageError(false)
}

useEffect(()=>{
    validateImage();
},[fileRejections])
```